tests/stdio/test_rename.c

```diff
@@ -124,6 +133,50 @@ void test() {
   err = rename("dir/subdir4/", "dir/subdir5/");
   assert(!err);
 
+  // Test renaming the same directory
+  err = rename("dir/file2", "dir/file2");
+  assert(!err);
+
+  // In Linux, renaming the root directory should return EBUSY.
+  // In the JS file system it reports EINVAL.
+  err = rename("/", "dir/file2");
+  assert(err == -1);
+#ifdef WASMFS
+  assert(errno == EBUSY);
+#else
+  assert(errno == EINVAL);
+#endif
+
+  err = rename("dir/file2", "/");
+  assert(err == -1);
+  assert(errno == ENOTEMPTY);
+
+  // Test renaming the current working directory while still root.
```

Comment:
It would be nice to have a separate test file just for testing interactions between `chdir`, `getcwd`, `rename`, and `unlink`. Maybe we can just leave a TODO for that in this PR?

Reply:
Sure, do you want me to remove `chdir` and `getcwd` then?